packages/medusa/src/api/auth/token/refresh/route.ts

```diff
@@ -0,0 +1,40 @@
+import { IAuthModuleService } from "@medusajs/types"
+import {
+  AuthenticatedMedusaRequest,
+  MedusaResponse,
+} from "../../../../types/routing"
+import {
+  ContainerRegistrationKeys,
+  ModuleRegistrationName,
+} from "@medusajs/utils"
+import { generateJwtTokenForAuthIdentity } from "../../utils/generate-jwt-token"
+
+// Retrieve a newly generated JWT token. All checks that the existing token is valid already happen in the auth middleware.
+// The token will include the actor ID, even if the token used to refresh didn't have one.
+// Note: We probably want to disallow refreshes if the password changes, and require reauth.
```

Comment:
I'll put this scenario in here, will leave it upto you to decide if its worth to tackle or not:

If a JWT token is compromised, anyone with that JWT can essentially create unlimited tokens here that will be valid using this endpoint here. We don't currently seem to have an invalidation strategy in place to prevent this.

Reply:
Yeah that's the nature of JWT tokens, and is the reason why they should be kept relatively short-lived. Although it might seem dangerous, losing JWT tokens is not something that happens that easily, so the practical risks are not as high as people make them sound.

When it comes to creating unlimited tokens, you can probably do enough damage with 1 as you can do with unlimited tokens, so it doesn't make the severity any higher.

Now, there are few strategies to deal with how you use JWT tokens, such as issuing longer-lived refresh tokens, or supporting invalidation, but I don't think it is something we have to focus on anytime soon. The only thing we might want to support is have a flag on the auth_identity to not allow for refreshes (eg. when you change your password), but even that is something that is for later on.

You could always use session storage if you cannot accept that risk, this is just for the cases where you want to use JWTs.

Reply:
> Although it might seem dangerous, losing JWT tokens is not something that happens that easily, so the practical risks are not as high as people make them sound.

This can be said about any security key 😄 

> When it comes to creating unlimited tokens, you can probably do enough damage with 1 as you can do with unlimited tokens, so it doesn't make the severity any higher.

Not exactly, you can blacklist one using a simple list where a human can simply add it. When you have a system that can generate unlimited tokens, its harder to blacklist with a simple manual intervention.

Either way, will resolve this as I just wanted to surface this.

Reply:
Yes indeed, security keys that are not handled manually (eg. API keys) have lower chance of being stolen. Not to say it doesn't happen, but it's all about tradeoffs. 

TBH, If you have a simple server, using sessions is probably easier to handle, so you can go with that. Once you get to a more distributed setup, session-less handling can be better (although if you add token invalidation, you get closer to session-based auth), so it's just about having either option available and making an informed decision.

You can always rotate the JWT secret (which is a bit of a big-bang approach), so there are other mechanisms to "invalidate" tokens, but I get your point 😄